data/virt_autotest/guest_unattended_installation_files/opensuse_tumbleweed_xen_pv_guest_x86_64.xml

```diff
@@ -174,5 +174,28 @@
       <username>root</username>
     </user>
   </users>
+  <scripts>
+    <!-- permit root login, password login and pubkeys login -->
+    <!-- /etc/ssh/sshd_config/sshd_config is not default installed in TW -->
+    <init-scripts config:type="list">
+      <script>
+        <source><![CDATA[
+sshd_config_file="/etc/ssh/sshd_config"
+if [ ! -f $sshd_config_file ]; then
+    echo -e "PermitRootLogin yes\nPubkeyAuthentication yes\nPasswordAuthentication yes" > $sshd_config_file
+else
+    keys="PermitRootLogin PubkeyAuthentication PasswordAuthentication"
+    for key in $keys; do
+        sed -i "/^[# ]*$key */{h;s/^[# ]*$key *.*\$/$key yes/};\${x;/^\$/{s//$key yes/;H};x}" $sshd_config_file
+    done
+fi
+[ -d /root/.ssh ] || mkdir /root/.ssh; chmod 700 /root/.ssh
+touch /root/.ssh/authorized_keys; chmod 600 /root/.ssh/authorized_keys
+echo "##Authorized-Keys##" >> /root/.ssh/authorized_keys
+]]>
+        </source>
+      </script>
+    </init-scripts>
+  </scripts>
```

Comment:
By the way, there seems to be more elegant way to enable root ssh login with public key. Have you tried it ?
https://doc.opensuse.org/projects/autoyast/#Configuration-Security-users-and-groups

Reply:
there is already the 'root' user and its auth keys in uefi guest autoyast control file, I coped that one and did not make change to <user> section. It worked in a previous snapshot but it did not take effect in the snapshot when sshd was changed. I guess the user creation procedure is prior to the sshd configuration files creation. Just my guess.

The autoyast document for opensuse Leap 15 seems to be outdated for TW. TW may be more like Leap 16.